File: Modul-6/Soal-4/PRAK604-2410817220012-NOOR-KHALISA.py
kode = input("Kode: ")
pesan = input("Pesan: ")
if len(kode) != len(pesan):
    print("Panjang kalimat berbeda, pesan palsu")
else:
    bintang = 0
    pagar = 0
    hasil =[]

    for i in range(len(kode)):
        if kode[i] == pesan[i]:
            hasil.append('*')
            bintang += 1
        else:
            hasil.append('#')
            pagar += 1
    print("".join(hasil))
    print(f"* = {bintang}")
    print(f"# = {pagar}")

    if bintang >= pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

File: Modul-6/Soal-4/PRAK604-2410817220012-NOOR-KHALISA.c
#include <stdio.h>
#include <string.h>
void cekPesan(char kode[], char pesan[]);
int main(){
    char kode[100], pesan[100];
    printf("Kode: ");
    fgets(kode, sizeof(kode), stdin);
    printf("Pesan: ");
    fgets(pesan, sizeof(pesan), stdin);
    kode[strcspn(kode, "\n")] = 0;
    pesan[strcspn(pesan, "\n")] = 0;
    cekPesan(kode, pesan);
    return 0;
}

void cekPesan(char kode[], char pesan[]){
    int panjangKode = strlen(kode);
    int panjangPesan = strlen(pesan);
    int bintang = 0, pagar = 0;
    char hasil[100];
    if (panjangKode != panjangPesan)
    {
        printf("Panjang kalimat berbeda, pesan palsu.");
        return;
    }
    for (int i = 0; i < panjangKode; i++)
    {
        if (kode[i] == pesan[i])
        {
            hasil[i] = '*';
            bintang++;
        }else
        {
            hasil[i] = '#';
            pagar++;
        }
    }
    hasil[panjangKode] = '\0';
    printf("%s\n", hasil);
    printf("* = %d\n", bintang);
    printf("# = %d\n", pagar);
    if (bintang >= pagar)
    {
        printf("Pesan Asli");
    }else
    {
        printf("Pesan Palsu.");
    }
}